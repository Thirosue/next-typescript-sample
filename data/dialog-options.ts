import React from 'react'

export type DialogOptions = {
  html?: boolean
  alert?: boolean
  icon?: 'info' | 'warn' | 'alert'
  title: string
  description: React.ReactNode
  confirmationText?: string
  cancellationText?: string
}
