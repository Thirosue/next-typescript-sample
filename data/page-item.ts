export type PageItem = {
  page: number
  totalPage?: number
  totalCount: number
  perPage: number
}
