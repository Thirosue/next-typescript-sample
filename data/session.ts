export type Session = {
  username?: string
  sub: string
  jwtToken: string
  email_verified?: boolean
}
