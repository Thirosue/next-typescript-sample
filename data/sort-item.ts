export type SortItem = {
  key: string
  order: '' | 'asc' | 'desc'
}
