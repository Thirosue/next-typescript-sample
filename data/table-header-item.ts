export type TableHeaderItem = {
  key?: string
  label: string
  sortable?: boolean
  onClick?: () => Promise<void>
}
