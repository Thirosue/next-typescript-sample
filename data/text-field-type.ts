export enum TextFieldType {
  Text = 'text',
  Number = 'number',
  Email = 'email',
  Password = 'password',
  Tel = 'tel',
  Url = 'url',
}
