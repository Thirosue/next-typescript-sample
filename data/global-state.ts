import { Session } from './session'

export type GlobalState = {
  session: Session
}
