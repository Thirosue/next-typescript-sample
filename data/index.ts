import { TextFieldType } from './text-field-type'

export { TextFieldType }
