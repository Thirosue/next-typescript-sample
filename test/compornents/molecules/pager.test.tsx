import React from 'react'
import { render } from '../../testUtils'
import { PageItem } from '../../../data/page-item'
import Pager from '../../../components/molecules/pager'

type Pages = {
  page1: Element
  page2: Element
  page3: Element
  page4: Element
  page5: Element
  page6: Element
  page7: Element
  page8: Element
  page9: Element
  page10: Element
}

class Page {
  container: HTMLElement
  constructor(container: HTMLElement) {
    this.container = container
  }

  一ページ目リンク(): Element {
    return this.container.querySelector('.page-link-1')
  }

  二ページ目リンク(): Element {
    return this.container.querySelector('.page-link-2')
  }

  三ページ目リンク(): Element {
    return this.container.querySelector('.page-link-3')
  }

  四ページ目リンク(): Element {
    return this.container.querySelector('.page-link-4')
  }

  五ページ目リンク(): Element {
    return this.container.querySelector('.page-link-5')
  }

  六ページ目リンク(): Element {
    return this.container.querySelector('.page-link-6')
  }

  七ページ目リンク(): Element {
    return this.container.querySelector('.page-link-7')
  }

  八ページ目リンク(): Element {
    return this.container.querySelector('.page-link-8')
  }

  九ページ目リンク(): Element {
    return this.container.querySelector('.page-link-9')
  }

  十ページ目リンク(): Element {
    return this.container.querySelector('.page-link-10')
  }

  ページリンク一覧(): Pages {
    return {
      page1: this.一ページ目リンク(),
      page2: this.二ページ目リンク(),
      page3: this.三ページ目リンク(),
      page4: this.四ページ目リンク(),
      page5: this.五ページ目リンク(),
      page6: this.六ページ目リンク(),
      page7: this.七ページ目リンク(),
      page8: this.八ページ目リンク(),
      page9: this.九ページ目リンク(),
      page10: this.十ページ目リンク(),
    }
  }
}

const handleClick = async (): Promise<void> => {}

describe('Pager components', () => {
  it('matches snapshot', () => {
    const item: PageItem = {
      page: 1,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    const { asFragment } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('全10ページ かつ 現在1ページ目を指定しているとき、ページャの数が期待値どおりであること', () => {
    const item: PageItem = {
      page: 1,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    const { container } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    const page = new Page(container)

    // 期待値
    // [1] 2 3 4 ... 10
    const { page1, page2, page3, page4, page5, page9, page10 } =
      page.ページリンク一覧()

    expect(page1).toBeTruthy()
    expect(page1.className).toContain('cursor-not-allowed')

    expect(page2).toBeTruthy()
    expect(page2.className).not.toContain('cursor-not-allowed')

    expect(page3).toBeTruthy()
    expect(page4).toBeTruthy()

    expect(page5).toBeNull()
    expect(page9).toBeNull()

    expect(page10).toBeTruthy()
    expect(page10.className).not.toContain('cursor-not-allowed')
  })

  it('全10ページ かつ 現在2ページ目を指定しているとき、ページャの数が期待値どおりであること', () => {
    const item: PageItem = {
      page: 2,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    const { container } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    const page = new Page(container)

    // 期待値
    // 1 [2] 3 4 5 ... 10
    const { page1, page2, page3, page4, page5, page6, page9, page10 } =
      page.ページリンク一覧()

    expect(page1).toBeTruthy()
    expect(page1.className).not.toContain('cursor-not-allowed')

    expect(page2).toBeTruthy()
    expect(page2.className).toContain('cursor-not-allowed')

    expect(page3).toBeTruthy()
    expect(page3.className).not.toContain('cursor-not-allowed')

    expect(page4).toBeTruthy()
    expect(page5).toBeTruthy()

    expect(page6).toBeNull()
    expect(page9).toBeNull()

    expect(page10).toBeTruthy()
    expect(page10.className).not.toContain('cursor-not-allowed')
  })

  it('全10ページ かつ 現在3ページ目を指定しているとき、ページャの数が期待値どおりであること', () => {
    const item: PageItem = {
      page: 3,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    const { container } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    const page = new Page(container)

    // 期待値
    // 1 2 [3] 4 5 6 ... 10
    const { page1, page2, page3, page4, page5, page6, page7, page9, page10 } =
      page.ページリンク一覧()

    expect(page1).toBeTruthy()
    expect(page1.className).not.toContain('cursor-not-allowed')

    expect(page2).toBeTruthy()
    expect(page2.className).not.toContain('cursor-not-allowed')

    expect(page3).toBeTruthy()
    expect(page3.className).toContain('cursor-not-allowed')

    expect(page4).toBeTruthy()
    expect(page4.className).not.toContain('cursor-not-allowed')

    expect(page5).toBeTruthy()
    expect(page6).toBeTruthy()

    expect(page7).toBeNull()
    expect(page9).toBeNull()

    expect(page10).toBeTruthy()
    expect(page10.className).not.toContain('cursor-not-allowed')
  })

  it('全10ページ かつ 現在4ページ目を指定しているとき、ページャの数が期待値どおりであること', () => {
    const item: PageItem = {
      page: 4,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    render(<Pager pageItem={item} search={handleClick} />, {})

    const { container } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    const page = new Page(container)

    // 期待値
    // 1 2 3 [4] 5 6 7 ... 10
    const {
      page1,
      page2,
      page3,
      page4,
      page5,
      page6,
      page7,
      page8,
      page9,
      page10,
    } = page.ページリンク一覧()

    expect(page1).toBeTruthy()
    expect(page1.className).not.toContain('cursor-not-allowed')

    expect(page2).toBeTruthy()
    expect(page2.className).not.toContain('cursor-not-allowed')

    expect(page3).toBeTruthy()
    expect(page3.className).not.toContain('cursor-not-allowed')

    expect(page4).toBeTruthy()
    expect(page4.className).toContain('cursor-not-allowed')

    expect(page5).toBeTruthy()
    expect(page5.className).not.toContain('cursor-not-allowed')

    expect(page6).toBeTruthy()
    expect(page7).toBeTruthy()

    expect(page8).toBeNull()
    expect(page9).toBeNull()

    expect(page10).toBeTruthy()
    expect(page10.className).not.toContain('cursor-not-allowed')
  })

  it('全10ページ かつ 現在5ページ目を指定しているとき、ページャの数が期待値どおりであること', () => {
    const item: PageItem = {
      page: 5,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    const { container } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    const page = new Page(container)

    // 期待値
    // 1 2 3 4 [5] 6 7 8 ... 10
    const {
      page1,
      page2,
      page3,
      page4,
      page5,
      page6,
      page7,
      page8,
      page9,
      page10,
    } = page.ページリンク一覧()

    expect(page1).toBeTruthy()
    expect(page1.className).not.toContain('cursor-not-allowed')

    expect(page2).toBeTruthy()
    expect(page2.className).not.toContain('cursor-not-allowed')

    expect(page3).toBeTruthy()
    expect(page3.className).not.toContain('cursor-not-allowed')

    expect(page4).toBeTruthy()
    expect(page4.className).not.toContain('cursor-not-allowed')

    expect(page5).toBeTruthy()
    expect(page5.className).toContain('cursor-not-allowed')

    expect(page6).toBeTruthy()
    expect(page6.className).not.toContain('cursor-not-allowed')

    expect(page7).toBeTruthy()
    expect(page8).toBeTruthy()

    expect(page9).toBeNull()

    expect(page10).toBeTruthy()
    expect(page10.className).not.toContain('cursor-not-allowed')
  })

  it('全10ページ かつ 現在6ページ目を指定しているとき、ページャの数が期待値どおりであること', () => {
    const item: PageItem = {
      page: 6,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    const { container } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    const page = new Page(container)

    // 期待値
    // 6ページ 1 ... 3 4 5 [6] 7 8 9 10
    const {
      page1,
      page2,
      page3,
      page4,
      page5,
      page6,
      page7,
      page8,
      page9,
      page10,
    } = page.ページリンク一覧()

    expect(page1).toBeTruthy()
    expect(page1.className).not.toContain('cursor-not-allowed')

    expect(page2).toBeNull()

    expect(page3).toBeTruthy()
    expect(page3.className).not.toContain('cursor-not-allowed')

    expect(page4).toBeTruthy()
    expect(page4.className).not.toContain('cursor-not-allowed')

    expect(page5).toBeTruthy()
    expect(page5.className).not.toContain('cursor-not-allowed')

    expect(page6).toBeTruthy()
    expect(page6.className).toContain('cursor-not-allowed')

    expect(page7).toBeTruthy()
    expect(page7.className).not.toContain('cursor-not-allowed')

    expect(page8).toBeTruthy()
    expect(page9).toBeTruthy()
    expect(page10).toBeTruthy()
    expect(page10.className).not.toContain('cursor-not-allowed')
  })

  it('全10ページ かつ 現在10ページ目を指定しているとき、ページャの数が期待値どおりであること', () => {
    const item: PageItem = {
      page: 10,
      totalPage: 10,
      totalCount: 97,
      perPage: 10,
    }
    const { container } = render(
      <Pager pageItem={item} search={handleClick} />,
      {}
    )
    const page = new Page(container)

    // 期待値
    // 10ページ 1 ... 7 8 9 [10]
    const {
      page1,
      page2,
      page3,
      page4,
      page5,
      page6,
      page7,
      page8,
      page9,
      page10,
    } = page.ページリンク一覧()

    expect(page1).toBeTruthy()
    expect(page1.className).not.toContain('cursor-not-allowed')

    expect(page2).toBeNull()
    expect(page3).toBeNull()
    expect(page4).toBeNull()
    expect(page5).toBeNull()
    expect(page6).toBeNull()

    expect(page7).toBeTruthy()
    expect(page7.className).not.toContain('cursor-not-allowed')

    expect(page8).toBeTruthy()
    expect(page8.className).not.toContain('cursor-not-allowed')

    expect(page9).toBeTruthy()
    expect(page9.className).not.toContain('cursor-not-allowed')

    expect(page10).toBeTruthy()
    expect(page10.className).toContain('cursor-not-allowed')
  })
})
