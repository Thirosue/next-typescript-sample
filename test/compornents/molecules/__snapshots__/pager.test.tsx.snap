// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager components matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white px-4 py-3 flex items-center justify-between border-t border-gray-200 sm:px-6"
  >
    <div
      class="flex-1 flex justify-between sm:hidden"
    >
      <a
        class="relative inline-flex items-center px-4 py-2 border border-gray-300 text-sm font-medium rounded-md text-gray-700 bg-white cursor-not-allowed"
        href="#"
      >
        Previous
      </a>
      <a
        class="relative inline-flex items-center px-4 py-2 border border-gray-300 text-sm font-medium rounded-md text-gray-700 bg-white hover:bg-gray-50"
        href="#"
      >
        Next
      </a>
    </div>
    <div
      class="hidden sm:flex-1 sm:flex sm:items-center sm:justify-between"
    >
      <div>
        <p
          class="text-sm text-gray-700"
        >
          <span>
            Showing
          </span>
          <span
            class="font-medium mx-1"
          >
            1
          </span>
          <span>
            to
          </span>
          <span
            class="font-medium mx-1"
          >
            2
          </span>
          <span>
            of
          </span>
          <span
            class="font-medium mx-1"
          >
            97
          </span>
          <span>
            results
          </span>
        </p>
      </div>
      <div>
        <nav
          aria-label="Pagination"
          class="relative z-0 inline-flex rounded-md shadow-sm -space-x-px"
        >
          <a
            class="relative inline-flex items-center px-4 py-2 border border-gray-300 text-sm font-medium rounded-md text-gray-700 bg-white cursor-not-allowed"
            href="#"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
          <a
            class="relative inline-flex items-center px-4 py-2 border text-sm font-medium page-link-1 z-10 bg-indigo-50 border-indigo-500 text-indigo-600 cursor-not-allowed"
            href="#"
          >
            1
          </a>
          <a
            class="relative inline-flex items-center px-4 py-2 border text-sm font-medium page-link-2 bg-white border-gray-300 text-gray-500 hover:bg-gray-50"
            href="#"
          >
            2
          </a>
          <a
            class="relative inline-flex items-center px-4 py-2 border text-sm font-medium page-link-3 bg-white border-gray-300 text-gray-500 hover:bg-gray-50"
            href="#"
          >
            3
          </a>
          <a
            class="relative inline-flex items-center px-4 py-2 border text-sm font-medium page-link-4 bg-white border-gray-300 text-gray-500 hover:bg-gray-50"
            href="#"
          >
            4
          </a>
          <span
            class="relative inline-flex items-center px-4 py-2 border border-gray-300 bg-white text-sm font-medium text-gray-700"
          >
            ...
          </span>
          <a
            class="relative inline-flex items-center px-4 py-2 border text-sm font-medium page-link-10 bg-white border-gray-300 text-gray-500 hover:bg-gray-50"
            href="#"
          >
            10
          </a>
          <a
            class="relative inline-flex items-center px-4 py-2 border border-gray-300 text-sm font-medium rounded-md text-gray-700 bg-white hover:bg-gray-50"
            href="#"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
